docs/Encoder.md

```diff
@@ -1,57 +1,81 @@
+<img src="images/amdc-encoder-input-highlighted.svg" />
 
-## PCB Layout
+More information on the D-sub connector can be found [here]( https://www.alliedelec.com/m/d/c3366066d9274ddf3c20bc3008518f5b.pdf?src-supplier=Allied+Electronics).
+
+### 3. Differential to Single-Ended Conversion
 
-All chips and d-sub connectors are located on the top layer. Signal routes from d-sub connectors to differential to single-ended converter and then to level shifter are all located on the top layer as well, while signals from level shifter to PicoZed are located in the intermediate level and each inverting and non-inverting pair of encoder signals are connected between each other by the termination resistor on the bottom layer. All signal traces have 6 mil thickness. Supply and ground traces connected to the chips have thickness of 10 mil.
+After passing through the D-sub connector, the differential ABZ signals (A, A̅, A, B̅, Z, Z̅) are converted to single-ended ABZ signals (OUT A, OUT B, OUT Z) using the [AM26C32](http://www.ti.com/lit/ds/symlink/am26c32.pdf) chip. This chip is supplied by `Vcc = 5 V DC` source from AMDC. The following table summarizes the absolute maximum ratings taken from the datasheet for this chip:
 
-It is recommended that bypass capacitors are used on the power supplies to reduce noise, and, if possible, short trace lengths should be used to avoid excessive loading.
+| Parameter                            | MIN    | MAX           |
+|--------------------------------------|--------|---------------|
+| Supply voltage Vcc                   |        | 7 V           |
+| Input voltage (A, A̅, A, B̅, Z or Z̅)   | -11 V  | 14 V          |
+| Differential input voltage           | -14 V  | 14 V          |
+| Output voltage (OUT A, OUT B, OUT Z) | -0.5 V | (Vcc + 0.5) V |
 
+Exposure to absolute maximum ratings for an extended period of time may affect device reliability. Therefore, it is recommended to operate under the recommended operating conditions:
 
+| Parameter                             | MIN   | NOM | MAX   |
+|---------------------------------------|-------|-----|-------|
+| Supply voltage Vcc                    | 4.5 V | 5 V | 5.5 V |
+| High-level input voltage (logic HIGH) | 2 V   |     | Vcc   |
+| Low-level input voltage (logic LOW)   | 0 V   |     | 0.8 V |
+| Common-mode input voltage             | -7 V  |     | 7 V   |
 
+When the chip is operated at nominal conditions (Vcc = 5 V, room temperature 25 C), the output signals of the chip are such that the high-level output voltage is 3.8 V or more and low-level output voltage is around 0.2 V. Typical current the chip consumes is about 10 mA. This translates to about 50 mW of power consumption with 5 V supply. The rise and fall times of the output signal are both approximately equal to 4 ns. Assuming the system as first order, the maximum frequency the chip can reliably operate at is approximately 0.35/4 ns = 87.5 MHz.

```

Comment:
> The rise and fall times of the output signal are both approximately equal to 4 ns. Assuming the system as first order, the maximum frequency the chip can reliably operate at is approximately 0.35/4 ns = 87.5 MHz

You have language like this in here in a couple places. Can you explain why we are including it? It sounds to me like an attempt to set a maximum frequency of encoder pulses that we are able to receive (which, please do specify!). However, I am not sure that this is the way to specify it.

Reply:
@elsevers Yes, I wanted to specify the maximum encoder pulse frequency that we are able to receive. I thought it is equal to the minimum of maximum frequencies each chip in the encoder chain can receive. I specified the maximum frequency for two chips (differential to single-ended converter and level shifter), but did not initially find the information about maximum frequency Picozed can receive from the encoder. Now I think I found some information about that, but I am not sure whether it is relevant (it is difficult to find where they explicitly say about maximum frequency):

- In the last paragraph in page 21 of [this document](http://zedboard.org/sites/default/files/documentations/5282-UG-PicoZed-7010_7020-v2_0.pdf), it says `Within a PL I/O Bank 34 or Bank 35, there are 50 I/O capable of up to 24 differential pairs per bank. Differential LVDS pairs on a -1 speed grade device are capable of 950Mbps of DDR data`. `Bank 35` corresponds to the pins where encoder signals are connected.
- In page 2 under `Feature Summary` table in [this document](https://www.xilinx.com/support/documentation/data_sheets/ds190-Zynq-7000-Overview.pdf), it says maximum frequency for the processing system is 667 MHz for -1 speed grade device and 766 MHz for -2 speed grade device. 

I am not sure whether these are relevant information and whether should be included. @npetersen2 Could you please help with this?

Reply:
@Anvar-Khamitov Thanks for looking into this. I believe the answer is as follows. This depends on the FPGA clock frequency.

The encoder data lines are read into the FPGA. Therefore they are sampled at the FPGA clock frequency. We run the FPGA at 200MHz (clk = 200MHz, Tclk = 1/200MHz = 5ns). I suggest we have a minimum of a 10x over-sampling ratio per encoder data line square wave voltage level. So, the frequency of the square wave on the encoder data line must be less than 1/(20*Tclk). This corresponds to a 10MHz square wave. This is the final answer.

The document should probably have some info related to encoder specs:
- Cycles per revolution (CPR)
- Pulses per revolution (PPR = 4 x CPR)

Let's do an example: For the 160kRPM project, this means Fm = 2.66k [Hz] mechanical revolutions per second. Let's assume the encoder is: CPR = 1024, PPR = 4096. Then, the square wave freq on the encoder data lines is Fm * PPR = ~11kHz.

If that math is right, it means that the ability of the FPGA is WAY higher than the encoder output... 10MHz >> 11kHz. So, I think a 1024 CPR encoder is fine for the 160kRPM project.

@elsevers @Anvar-Khamitov Can you guys check my math above? Does it make sense?

Reply:
In fact, if that math is right, I don't really think you can buy an encoder that wouldn't work with the AMDC.... All achievable encoder output rates are okay.

Reply:
@npetersen2 I think it should be Fm * PPR = ~11MHz instead of 11kHz in your calculations, which is higher than 10MHz.

Reply:
Oh excellent, that makes much more sense. I was pretty shocked at how small my number was!

Reply:
Okay:) thanks! I will then include in the document that 10MHz is the maximum frequency we are able to receive.